docs/keysgen/keysgen.go

```diff
@@ -0,0 +1,79 @@
+package keysgen
+
+import (
+	"fmt"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	bip39 "github.com/bitcoin-sv/go-sdk/compat/bip39"
+	chaincfg "github.com/bitcoin-sv/go-sdk/transaction/chaincfg"
+)
+
+// Keys represents a set of hierarchical deterministic (HD) keys,
+// including the extended private key (Xpriv), extended public key (Xpub),
+// and the mnemonic phrase used to generate these keys.
+type Keys struct {
+	Xpriv    string // The HD extended private key as a string.
+	Xpub     string // The HD extended public key as a string.
+	Mnemonic string // The mnemonic phrase used to derive the keys, if available.
+}
+
+// GenerateKeysFromString creates a new set of HD keys from an extended private key string.
+func GenerateKeysFromString(s string) (*Keys, error) {
+	xPriv, err := bip32.NewKeyFromString(s)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HD private key: %w", err)
+	}
+
+	xPub, err := xPriv.Neuter()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HD public key: %w", err)
+	}
+
+	return &Keys{Xpriv: xPriv.String(), Xpub: xPub.String()}, nil
+}
+
+// GenerateKeysFromMnemonic creates a new set of HD keys from a mnemonic phrase.
+func GenerateKeysFromMnemonic(mnemonic string) (*Keys, error) {
+	seed, err := bip39.NewSeedWithErrorChecking(mnemonic, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create seed: %w", err)
+	}
+
+	xPriv, err := bip32.NewMaster(seed, &chaincfg.MainNet)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HD private key: %w", err)
+	}
+
+	xPub, err := xPriv.Neuter()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HD public key: %w", err)
+	}
+
+	return &Keys{Xpriv: xPriv.String(), Xpub: xPub.String(), Mnemonic: mnemonic}, nil
+}
+
+// GenerateKeys generates a new set of HD keys using a randomly generated mnemonic phrase.
+func GenerateKeys() (*Keys, error) {
```

Comment:
This was tricky for us so I would propose to make:
1. RandomXPriv() function
2. RandomMnemonic() function 
... instead of this Generate Keys.

Besically because the things based on mnemonic are extremally slow (we experienced long lag during xpriv generation - even though we didn't need a mnemonic at all)

Reply:
Good point :) 

@chris-4chain please take a look at the following snippet.

```
// RandomXPriv generates a random extended private key (xPriv) for a given seed size.
// The seed size is specified as an unsigned 8-bit integer (n).
// It returns a pointer to the extended private key and an error if seed generation or
// the creation of the master node HD key fails.
func RandomXPriv(n uint8) (*bip32.ExtendedKey, error) {
	seed, err := bip32.GenerateSeed(n)
	if err != nil {
		return nil, fmt.Errorf("failed to generate seed: %w", err)
	}

	xPriv, err := bip32.NewMaster(seed, &chaincfg.MainNet)
	if err != nil {
		return nil, fmt.Errorf("failed to generate master node HD key: %w", err)
	}

	return xPriv, nil
}

// RandomMnemonic generates a mnemonic phrase consisting of words derived from the given bit size of entropy.
// The bit size must be a multiple of 32 and fall within the inclusive range of {128, 256}.
// It returns the mnemonic as a string and an error if entropy generation or mnemonic creation fails.
func RandomMnemonic(n int) (string, error) {
	entropy, err := bip39.NewEntropy(n)
	if err != nil {
		return "", fmt.Errorf("failed to generate entropy: %w", err)
	}

	mnemonic, err := bip39.NewMnemonic(entropy)
	if err != nil {
		return "", fmt.Errorf("failed to generate mnemonic: %w", err)
	}

	return mnemonic, nil
}
```

Reply:
Exactly 💯 👍 
But, apart from that (`RandomXPriv, RandomMnemonic`), additional function would be necessary:
```
XPrivFromMnemonic(mnemonic string) *bip32.ExtendedKey
XPrivFromString(xpriv string) *bip32.ExtendedKey
```

But check out what's implemented in the "old" go-client 
https://github.com/bitcoin-sv/spv-wallet-go-client/blob/e83ad92367a12124c1717ee57a0d3a78e290781c/xpriv/xpriv.go#L94-L133

Maybe the concept of `Key` struct and `KeyWithMnemonic` could be adopted.

Reply:
1. maybe the package should be called `walletkeys`
2. there should be 2 functions `Random` and `RandomWithMnemonic` which of course returns a little bit different structs (the best if the KeysWithMnemonic will extend Keys struct so it could be used in where only keys are needed, maybe with access to keys by methods
3. both of those functions shouldn't accept the arguments - we don't have any needs for arguments, those should be a helper functions to create keys in a default ways, to not copy this code everywhere. If someone would like to have better entropy, then he can create keys by himself.

Reply:
@dorzepowski  @chris-4chain  

I added missing functions, removed parameters from randoms, and used key structures as suggested. Please take a look and let me know if any changes are needed :).